controllers/userController.js

```diff
@@ -0,0 +1,71 @@
+import { Users } from '../models/Users.js';
+
+//create a new user
+export const createUser = async (req, res) => {
+  try {
+    const { firstName, lastName, email, password, role } = req.body;
+    const newUser = new Users({ firstName, lastName, email, password, role });
+    await newUser.save();
+    res
+      .status(201)
+      .json({ message: 'User created successfully', user: newUser });
+  } catch (error) {
+    res.status(400).json({ message: error.message });
+  }
+};
+
+// Get all users
+export const getAllUsers = async (req, res) => {
+  try {
+    const users = await Users.find();
+    res.status(200).json(users);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+};
+
+// Get a user by ID
+export const getUserById = async (req, res) => {
+  try {
+    const user = await Users.findById(req.params.id);
```

Comment:
The user's id is `ObjectId` so you must parse and validate logic here. Passing `1` will fail.